Small source code fixes/refactor

devilesk opened the pull request:
* Corrected errors with MaxClients and MAXPLAYERS.
* Corrected a couple instances of unclosed handles
* Renamed event handlers whose names conflict with l4d2util global forwards (OnRoundStart, OnRoundEnd, OnTankSpawn) in plugins that don't include l4d2util and hook the events themselves, otherwise they fire twice (from plugin's own hook + l4d2util global forward).
* Added l4d2util include to plugins that rely on its OnRoundStart global forward because they don't hook round_start themselves. This just makes the dependency on l4d2util explicit.

ProdigySim commented on addons/sourcemod/scripting/l4d_tank_control_eq.sp:
```diff
@@ -302,6 +306,8 @@ public chooseTank()
         }
         
+        CloseHandle(infectedTeam);
+        CloseHandle(infectedPool);
```
Looks like there's another leak in this file on line 301:

```pawn
h_whosHadTank = removeTanksFromPool(h_whosHadTank, teamSteamIds(L4D2Team_Infected));
```

That's another array being created in `teamSteamIds()` that never gets closed.

Might be a good to rename `teamSteamIds` to `createTeamSteamIdsArray`  so it looks more like something that creates a handle?

ProdigySim commented on addons/sourcemod/scripting/tankdoorfix.sp:
```diff
@@ -27,7 +27,7 @@
 new tankCount;
 
-new Float:nextTankPunchAllowed[19];
+new Float:nextTankPunchAllowed[MAXPLAYERS+1];
```
😭  nice

ProdigySim reviewed (approved):
Is there some documentation for the maxplayers/maxclients off-by-one errors? From Sourcemod's side? I asked around on SourceMod discord and didn't find any, so I [started a new article](https://wiki.alliedmods.net/Clients_(SourceMod_Scripting)).

Nice work cleaning these up; changes look good to me.

devilesk commented:
> Is there some documentation for the maxplayers/maxclients off-by-one errors? From Sourcemod's side? I asked around on SourceMod discord and didn't find any, so I [started a new article](https://wiki.alliedmods.net/Clients_(SourceMod_Scripting)).

I haven't seen any documentation from sourcemod. I learned about the nuances from reading threads about maxclients, maxplayers, and how to loop through entities on the alliedmods forum.

devilesk commented:
Fixed a leak caught by ProdigySim and did a little refactor so you can't use that function in a way that leaks like that. Also removed the l4d2util include that I added before, because it was preventing it from compiling. Instead I added a round_start hook and renamed the OnRoundStart handler to completely remove the dependency on l4d2util.
